api/v1beta1/types.go

```diff
@@ -893,6 +893,20 @@ func (s SubnetSpec) IsIPv6Enabled() bool {
 	return false
 }
 
+// GetSecurityRuleByDestination returns security group rule, which matches provided destination ports.
+func (s SubnetSpec) GetSecurityRuleByDestination(ports string) *SecurityRule {
+	if s.SecurityGroup.SecurityRules == nil {
```

Comment:
wrapping the below for loop in `if s.SecurityGroup.SecurityRules != nil {` is equivalent and allows us to remove the add'l `return nil` response (we don't need a specific `nil` return for this condition because the default return before exiting the function is `nil`)

Reply:
Should be fixed, thanks.